bin/src/hickory-dns.rs

```diff
@@ -455,10 +455,6 @@ fn run() -> Result<(), String> {
         listen_addrs.push(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
         listen_addrs.push(IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 0)));
     }
-    let _sockaddrs: Vec<SocketAddr> = listen_addrs
```

Comment:
Nit: to me, this is not an independent change (in the sense that reviewing the first commit left me wondering why you left these in), so I'd prefer having this squashed into the first commit.

Reply:
I can do that.  I separated them out because of [this](https://github.com/hickory-dns/hickory-dns/pull/2478#discussion_r1778470459)

Reply:
Yeah, maybe I was just wrong or I misunderstood the change or how/why these changes need to go together. Looking at the full diff again, it seems to me like the change to use the helper functions *might* be split from the change to switch to using the original socket addresses together, and that's what I was trying to convey in my earlier review. But maybe that just doesn't work/doesn't make sense for some reason that isn't obvious to me?

(Sorry for the misdirection...)